subsys/net/lib/zperf/zperf_shell.c

```diff
@@ -1195,6 +1195,7 @@ void zperf_shell_init(void)
 {
 	int ret;
 
+#if defined(MY_IP6ADDR_SET)
 	if (IS_ENABLED(CONFIG_NET_IPV6) && MY_IP6ADDR) {
```

Comment:
Better to do it like this
`if (IS_ENABLED(MY_IP6ADDR_SET) && IS_ENABLED(CONFIG_NET_IPV6) && MY_IP6ADDR) {`
so that the branch gets compile tested always.

Reply:
I thought `IS_ENABLED` is mainly for `CONFIG_` macros and not for any define?

Reply:
C pre-processor should not care how the symbol is defined. The `IS_ENABLED()` can be used instead of `#if defined()`

Reply:
Could we skip `IS_ENABLED(CONFIG_NET_IPV6)` perhaps if we add `IS_ENABLED(MY_IP6ADDR_SET)`? The latter is only set if IPv6 is enabled.

Reply:
> C pre-processor should not care how the symbol is defined. The `IS_ENABLED()` can be used instead of `#if defined()`

Sure, I was talking more about convention than C.


> Could we skip `IS_ENABLED(CONFIG_NET_IPV6)` perhaps if we add `IS_ENABLED(MY_IP6ADDR_SET)`? The latter is only set if IPv6 is enabled.

Sure.